Assignment2/Logger/FileLogger.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.IO;
+
+namespace Logger
+{
+    public class FileLogger : BaseLogger
+    {
+        public string LogFilePath { get; set; }
+
+        public FileLogger(){}
+        public FileLogger(string logFilePath)
+        {
+            LogFilePath = logFilePath;
+        }
+
+        public override void Log(LogLevel logLevel, string message)
+        {
+            var logMessage = DateTime.Now.ToString() + " " + ClassName + " " + logLevel + ": " + message + "\n";
+            if(!File.Exists(LogFilePath))
+            {
+                StreamWriter streamWriter = File.AppendText(LogFilePath);
+                streamWriter.Write(logMessage);
+                streamWriter.Close();
+                
+            }
+            else
+            {
+                throw new ArgumentException("File Path is null");
+            }
+        }
```

Comment:
This isn't the functionality we want here. If the file did exist, then an `ArgumentException` would be thrown, but we would want to append our log to the end of the file. Since `File.AppendText` creates a file if one doesn't exist, you would want something like this: 
```suggestion
            var logMessage = DateTime.Now.ToString() + " " + ClassName + " " + logLevel + ": " + message + "\n";

            if (!string.IsNullOrEmpty(LogFilePath))
            {
                File.AppendAllText(LogFilePath, logMessage);
            }
            else
            {
                throw new ArgumentNullException($"{nameof(LogFilePath)} can not be null or empty.");
            }
        }
```